Objects/typeobject.c

```diff
@@ -2108,6 +2108,7 @@ static int update_slot(PyTypeObject *, PyObject *);
 static void fixup_slot_dispatchers(PyTypeObject *);
 static int set_names(PyTypeObject *);
 static int init_subclass(PyTypeObject *, PyObject *);
+static int name_looks_like_slot(PyObject *name);
```

Comment:
If define it before use you will not need a forward declaration.

I would name it `is_dunder_name`.

Reply:
I've implemented it in that place because that keeps it close to the list of slot names list, so that it's less likely that the two get out of sync somehow.